src/simulators/statevector/qubitvector_thrust.hpp

```diff
@@ -0,0 +1,3146 @@
+/**
+ * This code is part of Qiskit.
+ *
+ * (C) Copyright IBM 2018, 2019.
+ *
+ * This code is licensed under the Apache License, Version 2.0. You may
+ * obtain a copy of this license in the LICENSE.txt file in the root directory
+ * of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Any modifications or derivative works of this code must retain this
+ * copyright notice, and modified files need to carry a notice indicating
+ * that they have been altered from the originals.
+ */
+
+
+
+#ifndef _qv_qubit_vector_thrust_hpp_
+#define _qv_qubit_vector_thrust_hpp_
+
+#include <cuda.h>
+#include <cuda_runtime.h>
```

Comment:
Should these be hidden in an #ifdef guard?

Reply:
I added ifdef QASM_THRUST to cover whole of qubitvector_thrust.hpp

Reply:
There may be a good reason why this cannot be done, but it's sort of unusual in code that uses templates to have if-then statements regarding float vs. double.  If there is a good reason for this (for example, some decision that is made, not just sizeof-dependent choices), leave it, of course, but it is a bit cluttered.